### DISCUSSION TITLE
yaml config array example

### DISCUSSION THREAD

**Discussion by bigsaam** [juftin/camply#76] [Q&A]
How would I structure the YAML config for multiple searches? i.e. unique recreation areas with different notification or search options

Thanks to everyone contributing to this proiect!

**Comment by juftin** [comment]
Hey @bigsaam camply isn't set up right now to perform multiple searches at a time, by that I mean you can't tell it to search in campground #1 in April and campground #2 in May. You can however pass an array in YAML for multiple recreation areas, campgrounds, or campsite Ids. You should be able to pass an array of notification providers on a single search too. Here's an example of multiple recreation areas: https://camply.readthedocs.io/en/latest/usage.html#using-a-yml-configuration-file-to-search-for-campsites

The Python API of camply currently handles arrays of search windows. Updating the YAML config would be easy to add that functionality in as well. 

Regarding multiple simultaneous searches, that might be a more involved refactor but I'll keep it in mind since it sounds really valuable. I'm currently using the docker image and separate containers to run simultaneous searches.

camply is starting to get more popular and will need some re-writing soon to make it easier to contribute to and develop new features down the line so I'll keep all of this in mind once I get started on that. I'll follow-up here once some progress is made. 

Glad you're enjoying camply! 🏕️

**Reply by juftin** [reply]
Leaving a multithreading note here for future @juftin regarding simultaneous searches:

```python
import concurrent
import datetime as datetime
import logging
from concurrent.futures import ThreadPoolExecutor

from camply.containers import SearchWindow
from camply.search import SearchRecreationDotGov
from dateutil.relativedelta import relativedelta

logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)

pool = concurrent.futures.ThreadPoolExecutor()
future_campsites = []
for i in range(6, 12):
    start = datetime.datetime(2022, i, 1)
    end = start + relativedelta(months=1)
    window = SearchWindow(start_date=start, end_date=end)
    search = SearchRecreationDotGov(search_window=window, recreation_area=2991)
    future_campsite = pool.submit(search.get_all_campsites)
    future_campsites.append(future_campsite)

futures = concurrent.futures.as_completed(future_campsites)
total_campsites = [campsite for future in futures for campsite in future.result()]
```
